File: video_1.5_loops/no_loops.py
from os import listdir

FAfiles = listdir('fasta')
FAfiles = ['fasta/' + fa for fa in FAfiles]
print(FAfiles)

def processFASTA(filename):
    with open(filename, mode = 'r') as mycon:
        mylines = mycon.readlines()
    mylines = [line.strip() for line in mylines]
    header = mylines[0]
    aaseq = "".join(mylines[1:])
    size = len(aaseq)
    return header, aaseq, size

FAoutput = [processFASTA(fa) for fa in FAfiles]

headers, AAseqs, sizes = zip(*FAoutput)

headers = list(headers)

#print(headers)
#print(AAseqs)
#print(sizes)

File: video_1.5_loops/no_loops.R
FAfiles = list.files("fasta")
FAfiles <- paste0("fasta/", FAfiles)

processFASTA <- function(filename){
  mylines = readLines(filename)
  aaseq = paste(mylines[-1], collapse = "")
  header = mylines[1]
  size = nchar(aaseq)
  return(list(header, aaseq, size))
}

#processFASTA(FAfiles[1])

FAoutput <- lapply(FAfiles, processFASTA)

headers <- sapply(FAoutput, function(x) x[[1]])
AAseqs <- sapply(FAoutput, function(x) x[[2]])
sizes <- sapply(FAoutput, function(x) x[[3]])
